File: dataset/ZeroShot/OPENAI_gpt-5-2025-08-07/Task13.py
def is_valid_walk(walk):
    if walk is None or len(walk) != 10:
        return False
    x = y = 0
    for d in walk:
        if not isinstance(d, str) or len(d) != 1:
            return False
        if d == 'n':
            y += 1
        elif d == 's':
            y -= 1
        elif d == 'e':
            x += 1
        elif d == 'w':
            x -= 1
        else:
            return False
    return x == 0 and y == 0

if __name__ == "__main__":
    tests = [
        ["n","s","n","s","n","s","n","s","n","s"],
        ["w","e","w","e","w","e","w","e","w","e","w","e"],
        ["w","w","w","w","w","w","w","w","w","w"],
        ["n","s","n","s","n","s","n","s","n","x"],
        ["n","n","e","e","s","s","w","w","n","s"],
    ]
    expected = [True, False, False, False, True]
    for i, t in enumerate(tests):
        print(f"Test {i+1}: {is_valid_walk(t)} (expected {expected[i]})")

File: dataset/ZeroShot - Copy/OPENAI_gpt-5-2025-08-07/Task13.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>
#include <stddef.h>

bool isValidWalk(const char* walk[], size_t len) {
    if (walk == NULL || len != 10) return false;
    int x = 0, y = 0;
    for (size_t i = 0; i < len; ++i) {
        const char* dir = walk[i];
        if (dir == NULL || strlen(dir) != 1) return false;
        char c = dir[0];
        switch (c) {
            case 'n': y += 1; break;
            case 's': y -= 1; break;
            case 'e': x += 1; break;
            case 'w': x -= 1; break;
            default: return false;
        }
    }
    return x == 0 && y == 0;
}

int main(void) {
    const char* t1[] = {"n","s","n","s","n","s","n","s","n","s"};
    const char* t2[] = {"w","e","w","e","w","e","w","e","w","e","w","e"};
    const char* t3[] = {"w","w","w","w","w","w","w","w","w","w"};
    const char* t4[] = {"n","s","n","s","n","s","n","s","n","x"};
    const char* t5[] = {"n","n","e","e","s","s","w","w","n","s"};

    bool expected[] = {true, false, false, false, true};
    const char** tests[] = {t1, t2, t3, t4, t5};
    size_t lengths[] = {10, 12, 10, 10, 10};

    for (size_t i = 0; i < 5; ++i) {
        bool result = isValidWalk(tests[i], lengths[i]);
        printf("Test %zu: %s (expected %s)\n", i + 1, result ? "true" : "false", expected[i] ? "true" : "false");
    }
    return 0;
}